docs/adr/0002-delivery-channel-creation-and-integration.md

```diff
@@ -0,0 +1,85 @@
+2. Store user preferences somewhere that is accessible to the management api, then when a new notification with a certain notification type and notification severity comes through, check what type of communication types  the user has 'subscribed' to and then send a message through the delivery channel.
+
+The latter option was chosen as it provides a user the ability to choose what notifications they want out of the box, if the former is chosen, all users/businesses integrating with this service will then need to write their own notifications user management service.
+
+User settings for which notifications will go to what communication types for each user will be stored inside a User Notifications Settings table which will be accessed by the Management Api. Configuration in this table will include the Notification Type and the appropriate communication types associated to that notification type. This table will have an associated crud api that will allow a user to view/change the stored settings.
+
+### Management Api and usable Delivery Channels
+
+The management API will host configuration per tenant for what communication types can be sent through which delivery channels in the following format:
+
+```json
+[
+  {
+    "id": "dc1",
+    "displayName": "Airship",
+    "communicationTypes": ["email", "sms", "web-push"]
+  }, 
+  {
+    "id": "dc2",
+    "displayName": "Twilio",
+    "communicationTypes": ["sms"]
+  }
+]
+```
+where 'communicationTypes' specifies what communication types are supported in this Delivery Channel. This allows for consumers of Marain.UserNotifications to specify if they want to only use one kind communication type from a certain Delivery Channel that supports multiple communication types, eg: an organisation using `Marain.UserNotifications` can specify that they want to use sms's from twilio and only communication types from that delivery channel.
+
+A delivery channel will have a generic api that can be consumed by the Management Api and be able to transform data provided in the Notification into a format that is specific to that communication type (this will be discussed in a sequential adr). The result of this call will be a callback uri/null depending on how the delivery channel has been implemented.
+
+### Delivery Channel configuration 
+
+A delivery channel will also need the configuration of the underlying platform to be stored somewhere (api keys, secrets, etc). This will be stored in a keyvault. If a delivery channel needs to be used, modifying the keyvault with the correct key and updating the configuration for the delivery channel in the management api to specify which communication types will be used. 
```

Comment:
Bear in mind that `Marain.UserNotifications` is a multi-tenanted service. Per-tenant configuration is attached to the tenant - we currently only do this with storage configuration, but I think the delivery channel configuration will need to be stored in this way as well.

That doesn't preclude the actual data being stored in KeyVault - for example, [BlobStorageConfiguration](https://github.com/corvus-dotnet/Corvus.Tenancy/blob/master/Solutions/Corvus.Azure.Storage.Tenancy/Corvus/Azure/Storage/Tenancy/BlobStorageConfiguration.cs) allows for either a full connection string to be provided in the `AccountName` property, or for the account secret to be retrieved from a KeyVault by specifying the `KeyVaultName` and `AccountKeySecretName `. We could do something similar for these more generic settings.

Reply:
I have updated this section to be more explicit.